kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationIT.java

```diff
@@ -94,8 +94,10 @@ class OauthBearerValidationIT {
     private static OauthServerContainer oauthServer;
 
     @BeforeAll
-    public static void beforeAll() {
-        // required to permit Kafka to interact with our endpoint.
+    static void beforeAll() {
+        // Kafka 4.0 requires that the org.apache.kafka.sasl.oauthbearer.allowed.urls sys property is set in order to use Oauth Bearer.
+        // The Kafka Broker and Proxy requires that JWKS_ENDPOINT_URL is in the allow list.
+        // The Kafka Client requires that TOKEN_ENDPOINT_URL is in the allow list.
         System.setProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG, JWKS_ENDPOINT_URL + "," + TOKEN_ENDPOINT_URL);
```

Comment:
This is a bit of a compromise - in two ways:

* It'd be much better if Kafka didn't rely on a system property for configuring the allow list, or at least had separate client and server side properties.   Maybe an RFE against Kafka would be a good idea.  That would allow the test to be much more precise.

* Separately, for this IT test case, the Kafka Broker needs the JWKS_ENDPOINT_URL in allow list too.   Ideally, we'd use a Container based Kafka so we had precise control, but we've got no way to influence the system properties for the Container's JVM.

So, we sledgehammer it - and set both Token and JWKS endpoints from the IT test.  I've written the unit test in such a way to give us confidence that Kroxylicious will do the right thing.